### PR TITLE
chore: added 3779 program

### PR DESCRIPTION
**Pull request by Peedee2002**
this is a little big, to make sure I also pass tests. Also, I reran the scraper. Here is 3779 running!

~~no, this is not because of that one dude whining on discord~~

image

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -14,5 +13,6 @@
     "python.linting.mypyEnabled": true,
     "python.linting.mypyArgs": [
         "--config-file=${workspaceFolder}/backend/mypy.ini"
-    ]
+    ],
+    "python.analysis.typeCheckingMode": "basic"
```
Comment (imagine-hussain) [inline .vscode/settings.json:17]: this is pylance right?

**Review by imagine-hussain — APPROVED** [review]
looks good

File: .vscode/settings.json (hunk 1 above)
Comment (Peedee2002) [reply .vscode/settings.json:17]: It's the built-in type checker, idk what it is 👀

**Comment by Peedee2002** [comment]
> looks good

Btw I don't have perms, you will need hit merge yourself

**Pull request merged**